modules/sc-mesh-secure-deployment/src/nats/src/cbma_adaptation.py

```diff
@@ -398,21 +491,20 @@ def __update_cbma_interface_lists(self) -> None:
             if interface in self.__lower_cbma_interfaces:
                 self.__lower_cbma_interfaces.remove(interface)
 
-    @staticmethod
-    def __wait_for_ap(timeout: int = 4) -> bool:
+    def __wait_for_ap(self, timeout: int = 4) -> bool:
         start_time = time.time()
         while True:
             try:
-                result = subprocess.check_output(["iw", "dev", "wlan1", "info"])
-                result_str = result.decode("utf-8")
-                # Use regular expressions to extract the interface type
-                match = re.search(r"type\s+([\w-]+)", result_str)
-                if match.group(1) == "AP":
+                result = subprocess.check_output(
+                    ["iw", "dev", "wlan1", "info"]
+                ).decode()
```

Comment:
With `text=True` in `check_output` you don't need to specify the `.decode()` :dango:

Reply:
but then 'if "type AP" in result' wouldn't work but would require b"type AP" which doesn't look nice. Change would also affect unit test's expected result check thus I'd rather keep it as is.